packages/x-date-pickers/src/internals/hooks/useDesktopPicker/useDesktopPicker.tsx

```diff
@@ -129,6 +151,36 @@ export const useDesktopPicker = <
 
   const handleInputRef = useForkRef(inputRef, fieldProps.inputRef);
 
+  let labelledById = labelId;
+  if (isToolbarHidden) {
+    if (label) {
+      labelledById = `${labelId}-label`;
+    } else {
+      labelledById = undefined;
+    }
+  }
+  if (
+    !labelledById &&
+    // ts complains about unknown prop access without casting
+    !(innerSlotProps?.popper as any)?.['aria-labelledby'] &&
+    process.env.NODE_ENV !== 'production' &&
```

Comment:
Try to set `if(process.env.NODE_ENV !== 'production')` and then another if inside with the other consitions.
That way, the `if` code is removed in production.

Reply:
Should our build process do it? 🤔 
It does not seem to happen after I've tested it. 🙈

Reply:
Or did you only mean that it could by an option if someone would choose so?
Because yeah, I think we ought to ship it so that others writing tests would also be able to see this warning pop up.

Reply:
I think it's usually people's build step which removes all the `if(process.env.NODE_END !== 'production') {}` because it is transformed into `if(false) {}` which is automatically removed.

The change I propose (and that we are applying on our others dev-only warnings) it just here to allow their build step to strip it correctly :+1:

Reply:
Yeah, was thinking about just that. 👌 
Will update, thanks for spotting it. 😉